### PR TITLE
Fix random crashing when placing and throwing vehicles.

### PR DESCRIPTION
**Pull request by Phylogeny**
So, I was able to reproduce the crashing upon placing vehicles many times, and even reproduced it once upon throwing one. Well, famous last words, but I believe Ive fixed it. I don't believe tasks scedualed with FML's server instance run on the main thread. In any event, I believe the problem was scheduling a task (where an entity is spawned in the world) there, rather than with the world's server instance. Only time will tell.

Also, I fixed two separate bugs that each caused vehicles to be lost on very rare occasion when picking up and placing them _many_ times in a row very quickly.

Besides that, I fixed a few minor typos/issues, allowed raytraced vehicles to be picked up, and allowed packets to be registered more efficiently and on both sides, while I was at it.

### CODE REVIEW

**Comment by MrCrayfish** [comment]
Thank you for this PR. This will be merged for next week's update.

**Comment by Phylogeny** [comment]
No problem.

**Comment by Phylogeny** [comment]
I've also just allowed raytraced vehicles to be left-click attacked, as well as a config to disable that feature.

**Comment by MrCrayfish** [comment]
Can you check out the review I started about the refactored PacketHandler?

**Comment by Phylogeny** [comment]
I can't seem to find it. Can you direct me to it?

File: src/main/java/com/mrcrayfish/vehicle/network/PacketHandler.java [hunk 1]
```diff
@@ -7,28 +7,28 @@
 
 public class PacketHandler
 {
-	public static final SimpleNetworkWrapper INSTANCE = NetworkRegistry.INSTANCE.newSimpleChannel(Reference.MOD_ID);
-	private static int messageId = 0;
+    public static final SimpleNetworkWrapper INSTANCE = NetworkRegistry.INSTANCE.newSimpleChannel(Reference.MOD_ID);
+    private static int messageId = 0;
 
-	private static enum Side
+    private static enum Side
```
[the remark this answers is not in the document]
Comment (Phylogeny) [reply src/main/java/com/mrcrayfish/vehicle/network/PacketHandler.java]: There we go. Ok, I'll take a look.

[the remark this answers is not in the document]
Comment (Phylogeny) [reply src/main/java/com/mrcrayfish/vehicle/network/PacketHandler.java]: Here's an example: https://github.com/Phylogeny/ExtraBitManipulation/blob/1.12/src/main/java/com/phylogeny/extrabitmanipulation/packet/PacketChangeGlOperationList.java

[the remark this answers is not in the document]
Comment (Phylogeny) [reply src/main/java/com/mrcrayfish/vehicle/network/PacketHandler.java]: If you handle things in a way that such a 2-to-1 server/client packet consolidation just isn't ever relevant (or you just don't want to do it), that's fine; I can remove it.

[the remark this answers is not in the document]
Comment (MrCrayfish) [reply src/main/java/com/mrcrayfish/vehicle/network/PacketHandler.java]: Okay, that makes sense. I've personally never needed them on both sides but it's no harm having it.

[the remark this answers is not in the document]
Comment (Phylogeny) [reply src/main/java/com/mrcrayfish/vehicle/network/PacketHandler.java]: Sounds good.

**Comment by MrCrayfish** [comment]
Alright, is this ready to be merged?

**Comment by Phylogeny** [comment]
Yes, I do believe it is.

**Pull request merged**